docs/configdoc.md

```diff
@@ -0,0 +1,98 @@
+|kf-poll-interval|`"30s"`|Periodic interval at which to scrape metrics using Prometheus| Yes |
+|n-metric-samples|`"20"`|Number of Metric Samples| Yes |
+
+
+## [xdp-root-program]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|name|`"xdp-root"`|Name of subdirectory in which to extract artifact| Yes |
+|artifact|`"xdp-root.tar.gz"`|Filename of xdp-root package. Only tar.gz and .zip formats are supported| Yes |
+|ingress-map-name|`"root_array"`|Ingress map name of xdp-root program| Yes |
+|command|`"xdp_root"`|Command to run xdp-root program| Yes |
+|version|`"1.01"`|Version of xdp-root program| Yes |
+|user-program-daemon|`"false"`|Set to true it requires l3afd to stop the application (via SIGTERM on Linux or SIGKILL on Windows)| Yes |
+
+## [tc-root-program]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|name|`"tc_root"`|Name of subdirectory in which to extract artifact| Yes |
+|artifact|`"l3af_tc_root.tar.gz"`|Filename of tc_root package| Yes |
+|ingress-map-name|`"tc_ingress_root_array"`|Ingress map name of tc_root program| Yes |
+|egress-map-name|`"tc_egress_root_array"`|Egress map name of tc_root program,for more info about ingress/egress check [cilium](https://docs.cilium.io/en/v1.9/concepts/ebpf/intro/)| Yes |
+|command|`"tc_root"`|Command to run tc_root program| Yes |
+|version|`"1.0"`|Version of tc_root program| Yes |
+|user-program-daemon|`"false"`|Boolean to check xdp-root is user-program daemon or not| Yes |
+
+## [l3af-configs]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|restapi-addr|`"localhost:53000"`| Hostname and Port of l3af-configs REST API | Yes |
+
+# [l3af-config-store]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|filename|`"/etc/l3afd/l3af-config.json"`|Absolute path of persistent config file where we are storing L3afBPFPrograms objects. For more info see [models](https://github.com/l3af-project/l3afd/blob/main/models/l3afd.go)| Yes |
+
+# [mtls]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|enabled| `"true"` | Boolean to check mtls enabled or not on REST API exposed by l3afd| Yes |
+|min-tls-version|`"1.3"`| Minimum tls version allowed| No |
+|cert-dir|`"/etc/l3af/certs"`|Absolute path of ca certificates. In Linux Context it is pointing to a filesystem directory, but in Windows it can point to [certificate store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores) | Yes |
```

Comment:
```suggestion
|cert-dir|`"/etc/l3af/certs"`|Absolute path of ca certificates. On Linux this points to a filesystem directory, but on Windows it can point to [certificate store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores) | Yes |
```